odoo/models.py

```diff
@@ -4415,9 +4415,28 @@ def _load_records(self, data_list, update=False):
                 imd.browse(d_id).unlink()
                 to_create.append(data)
 
+        def must_update(rec, fname, val):
+            """ Determine whether ``rec[fname]`` must be updated with ``val``. """
+            field = rec._fields.get(fname)
+            if field is None:
+                return True
+
+            if val is None or val is False:
+                return not rec[fname]
+
+            if field.type in ('many2many', 'one2many'):
+                return field.convert_to_cache([], rec) != field.convert_to_cache(val, rec)
```

Comment:
By the way, the left-hand expression is 100% equivalent to `field.convert_to_cache(rec[fname], rec)`.  Which means that both the `if` and `else` clauses can be identical.  And I believe the `if val is None...` above is useless too, as most methods `field.convert_to_cache()` contain similar code.